src/compiler/reverse.jl

```diff
@@ -321,6 +321,18 @@ function adjoint(pr::Primal)
         continue
       end
     end
+
+    # This is corresponds to a catch blocks which technically
+    # has predecessors but they are not modelled in the IRTools CFG.
+    # We put an error message at the beginning of said block.
+    if has_leave && isempty(predecessors(b)) && b.id != 1
+        _, f_stmt = first(b)
+        li = pr.ir.lines[f_stmt.line]
+        li = LineNumberNode(Int(li.line), li.file)
+        pushfirst!(rb, stmt(xcall(Base, :error,
+                                  "Can't differentiate function execution in catch block at $(li).")))
+    end
```

Comment:
I don't think throwing the error in the catch block is right.
I think we need to put it at the `throw` location.
Because I assume we also do not support `finally` blocks?

Reply:
Try finally is implemented in term of `enter` and `leave` which can roughly be mapped to the following pseudo code:

```julia
try
  foo()
  origin = 1
catch
  origin = 2
end

if origin == 2
  rethrow()
end
```

I will improve the error message to include try/finally as another construct which is not supported when an error is thrown in the block.